dartagnan/src/main/java/com/dat3m/dartagnan/program/Function.java

```diff
@@ -59,7 +59,8 @@ public Function(String name, FunctionType type, List<String> parameterNames, int
 
     @Override
     public Type getType() {
-        return TypeFactory.getInstance().getArchType();
+        final TypeFactory types = program.getEventFactory().getExpressionFactory().getTypeFactory();
+        return types.getArchType();
```

Comment:
Usages like this suggest that `Program` should probably provide direct getters for the different factories.